### ISSUE TITLE
[1.12.2] Spawn rates with Animania 1.3.4 on server are way too high

### ISSUE THREAD

**Issue by twoworlds** [capnkirok/animania#378]
Hi,

I updated both server and client to animania 1.3.4

After playing for an hour I found this : https://imgur.com/a/SCg1x

Cows everywhere where the population of chickens and pigs are as expected. I also found, that the sheep aren't spawning babies any more.

**Comment by Gepett0** [comment]
Additional to my issue #377 I'm excepting the cowpocalypse too.
Had to switch to creative to get along with this bull-shit ;-)

**Comment by capnkirok** [comment]
Hey both -

I've tested and tested this, so I know spawn control is working. Here's what I recommend. Please both try the below and get back to me. 

- Lower your Spawn Limit for Cows to say 30
- Increase your gestation timer to say 48000 (it should be 21000). This makes a full two Minecraft days for pregnancy.
- Restart your server
- Cull (as in kill) the excess animals
- If you really feel like it, you can separate the Cows and the Bulls. 
- Let me know the results

**Comment by twoworlds** [comment]
I changed the settings and restarted the server. I will report what happend

**Comment by D3nnis3n** [comment]
Until the recent patch i had Mobs aggregating in two or three chunks. It was devastating. No grass left, no water left (no infinite water on my Server, despite in oceans and rivers biomes). With the 1.3.4 Update they finally spread out over the world, but i still think it's too much Mobs. Just as a Feedback.

**Comment by capnkirok** [comment]
Thanks, I'm going to lower them even further. You can also do this manually, by just changing spawn limits.

**Comment by twoworlds** [comment]
So with this altered settings it seems, that no animal is breeding at all. I ran the client for 8 hours but neither cows, pigs nor chickes breed. When I leave the chunk, some animals disappear, as already reported here.

**Comment by Gepett0** [comment]
Hi,

just an idea, don't know if even possible, cause I'm nothing near in JAVA or MC modding:

1. Every animal gets an NBT-tag with its owner + a family"name". (possible???)
2. Per cfg(?) you can define how much animals/families one player can get per sort.
   e.g. A herd cows with 2 bulls, 10 Cows and 5 calfs.
3. If you exceed the limit e.g. you got a calf growing to a bull, than 1 bull dies "cause of age" :-)
   Or just simple they don't breed anymore, like in real life if the critical mass is achieved, except
   for the human race........

BR

**Comment by capnkirok** [comment]
@twoworlds thank you for the testing! Yes, that is what I see. Spawning is under control, that part is working. And if you were at your limit, then breeding doesn't work as the animals would have been removed at birth, or just take forever, especially if they aren't fed. I'm glad you get the same result.

Animals disappearing is another issue, which I am working on now. That should not be happening. It turns out that Minecraft's default age is set to 0 after restart, so it thinks existing animals have just spawned (which means my code thinks they should be removed at restart if you are over your limit). I am mystified why MC did things that way. I can't rely on mc's age timer.

Two changes I am making:
- The mother will stay pregnant until there is room under the limit OR if a player is nearby. This prevents births over the limit in nearby ticking chunks, and isn't as brutal as the current method as the mother stays pregnant.
- Once an animal is spawned in the world it will get a starting age. No animal that has an age will ever be removed (until I implement death by natural causes).

**Comment by Gepett0** [comment]
Did some more testing.
Killed all calfs and bulls 1 1/2 hours ago, only 5 cows were left, some may have been pregnant (couldn't check because of no status shown in HWYLA).
Now its cowpocalypse inc:
https://picload.org/view/dgigiwaw/2017-10-09_20.11.21.png.html

Strangely my sheeps haven't got any newborns since yesterday.
Could you prevent spawning of cows outside of the enclosure somwhow?

For more information config file: https://pastebin.com/thmEJDCi
Only changed plantsRemovedAfterEating=false and waterRemovedAfterDrinking=false as far as I remember.

**Comment by capnkirok** [comment]
@Gepett0 you still have the old config value of 60 for Spawn Limit Cows.

![image](https://user-images.githubusercontent.com/9628379/31353250-1498a85a-ad32-11e7-91fa-744c9775596c.png)

**Comment by Gepett0** [comment]
Whats the minimum distance for cows and bulls to not breed?

**Comment by capnkirok** [comment]
Within 3 blocks if unmated. So, you can't just separate by a single fence (true in RL too).

**Comment by capnkirok** [comment]
TL;DR
Ok, I've changed (once again), how spawning works, and lowered the limits. Fix rolling out tonight. Thanks everyone for the input.
